### PR TITLE
Ref(SPV-1216) Fix unit tests status codes

### PR DESCRIPTION
**Pull request by ac4ch**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: internal/api/v1/user/configs/configs_api_test.go [hunk 1]
```diff
@@ -60,7 +60,12 @@ func TestConfigsAPI_SharedConfig_APIResponses(t *testing.T) {
 
 			// then:
 			got, err := wallet.SharedConfig(context.Background())
-			require.ErrorIs(t, err, tc.expectedErr)
```
Comment (chris-4chain) [inline internal/api/v1/user/configs/configs_api_test.go:63]: Why not ErrorIs ? Doesn't work that way?

Comment (ac4ch) [reply internal/api/v1/user/configs/configs_api_test.go:63]: with contains we have this 
```shell 
=== RUN   TestAccessKeyAPI_RevokeAccessKey
=== RUN   TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_str_response:_500
--- PASS: TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_str_response:_500 (0.01s)
=== RUN   TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_200
--- PASS: TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_200 (0.00s)
=== RUN   TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_400
--- PASS: TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_400 (0.00s)
--- PASS: TestAccessKeyAPI_RevokeAccessKey (0.02s)
PASS
ok      github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/users   1.062s
```
with errorIs we getting on same Test but with just the asserting change to ErrorIs
```shell
=== RUN   TestAccessKeyAPI_RevokeAccessKey
=== RUN   TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_200
--- PASS: TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_200 (0.01s)
=== RUN   TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_400
--- PASS: TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_response:_400 (0.00s)
=== RUN   TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_str_response:_500
    /Users/augustynchmiel/4chain/spv/spv-wallet-go-client-v2/internal/api/v1/user/users/access_key_api_test.go:216:
                Error Trace:    /Users/augustynchmiel/4chain/spv/spv-wallet-go-client-v2/internal/api/v1/user/users/access_key_api_test.go:216
                Error:          Target error should be in err chain:
                                expected: "unrecognized response from API"
                                in chain: "failed to send HTTP DELETE request to revoke access key with ID: 081743f7-040e-45a3-8c36-dde39001e20d via User Access Keys API: HTTP response failure: unrecognized response from API: unexpected internal server failure"
                                        "HTTP response failure: unrecognized response from API: unexpected internal server failure"
                                        "unrecognized response from API: unexpected internal server failure"
                                        "unrecognized response from API"
                Test:           TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_str_response:_500
--- FAIL: TestAccessKeyAPI_RevokeAccessKey/HTTP_DELETE_/api/v1/users/current/keys/081743f7-040e-45a3-8c36-dde39001e20d_str_response:_500 (0.00s)
--- FAIL: TestAccessKeyAPI_RevokeAccessKey (0.02s)
FAIL
FAIL    github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/users   0.818s
```

Comment (ac4ch) [reply internal/api/v1/user/configs/configs_api_test.go:63]: @chris-4chain if you prefer we could use also `require.ErrorContains(t, err, tc.expectedErr.Error())`

**Comment by mgosek-4chain** [comment]
Here's the solution how we can simplify error checks by using only errors.Is. 

https://github.com/4chain-ag/spv-wallet-go-client-v2/pull/26

**Comment by mgosek-4chain** [comment]
There is still one thing to improve which I forgot to do in my PR:

1. Add such stub response per each testpkg:

```
func NewInternalServerSPVError() models.SPVError {
	return models.SPVError{
		Message:    http.StatusText(http.StatusInternalServerError),
		StatusCode: http.StatusInternalServerError,
		Code:       models.UnknownErrorCode,
	}
}
```

2. Per each unit test add case where SPVError JSON is returned for 500 HTTP status code 

For ex.

```
	"HTTP GET /api/v1/configs/shared response: 500": {
			expectedErr: configstest.NewInternalServerSPVError(),
			responder:   httpmock.NewJsonResponderOrPanic(http.StatusInternalServerError, configstest.NewInternalServerSPVError()),
		},
```

**Review by mgosek-4chain — APPROVED** [review]

**Pull request merged**